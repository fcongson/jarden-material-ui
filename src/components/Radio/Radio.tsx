import styled from '@emotion/styled'
import { FormControlLabel, Radio as MaterialRadio, ThemeProvider } from '@material-ui/core'
import * as React from 'react'
import { palette, transition } from '../../styles'
import { theme } from '../../styles/material'

interface RadioProps {
  label?: string
  value?: any
  disabled?: boolean
}

export const Radio: React.FC<RadioProps> = ({ label, value, disabled, ...rest }) => (
  <ThemeProvider theme={theme}>
    <FormControlLabel
      label={label}
      value={value}
      disabled={disabled}
      control={<MaterialRadio icon={<RadioIcon />} checkedIcon={<RadioIcon checked />} />}
      {...rest}
    />
  </ThemeProvider>
)

const RadioIcon = styled.span((props: { checked?: boolean }) => ({
  height: 32,
  width: 32,
  backgroundColor: props.checked ? palette['white'] : palette['grey-base'],
  border: props.checked ? `8px solid ${palette['turquoise']}` : '2px solid transparent',
  borderRadius: '50%',
  boxSizing: 'border-box',
  transition: transition,
  'input:hover ~ &': {
    backgroundColor: palette['white-base'],
    border: props.checked ? `8px solid ${palette['turquoise']}` : `2px solid ${palette['turquoise']}`
  },
  'input:disabled ~ &': {
    backgroundColor: palette['grey-base'],
    border: '8px solid transparent'
  }
}))
