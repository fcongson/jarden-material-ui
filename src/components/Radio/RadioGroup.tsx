import { RadioGroup as MaterialRadioGroup, ThemeProvider } from '@material-ui/core'
import * as React from 'react'
import { theme } from '../../styles/material'

interface RadioGroupProps {
  name?: string
  value?: any
  onChange: (value: any) => void
}

export const RadioGroup: React.FC<RadioGroupProps> = ({ name, value, onChange, children, ...rest }) => (
  <ThemeProvider theme={theme}>
    <MaterialRadioGroup
      name={name}
      value={value}
      onChange={event => {
        if (!!onChange) onChange(event.target.value)
      }}
      {...rest}>
      {children}
    </MaterialRadioGroup>
  </ThemeProvider>
)
