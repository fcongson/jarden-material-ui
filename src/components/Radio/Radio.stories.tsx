import * as React from 'react'
import { Radio } from './Radio'
import { RadioGroup } from './RadioGroup'
import { useState } from 'react'

export default {
  title: 'Radio',
  component: Radio
}

export const Default = () => {
  const [value, setValue] = useState()
  return (
    <RadioGroup name='group' value={value} onChange={value => setValue(value)}>
      <Radio label='Lorem' value='Lorem' />
      <Radio label='Ipsum' value='Ipsum' />
      <Radio label='Dolor' value='Dolor' />
      <Radio label='Sit' value='Sit' />
      <Radio label='Amet' value='Amet' />
    </RadioGroup>
  )
}

export const Disabled = () => {
  const [value, setValue] = useState()
  return (
    <RadioGroup name='group' value={value} onChange={value => setValue(value)}>
      <Radio label='Lorem' value='Lorem' disabled />
      <Radio label='Ipsum' value='Ipsum' disabled />
      <Radio label='Dolor' value='Dolor' disabled />
      <Radio label='Sit' value='Sit' disabled />
      <Radio label='Amet' value='Amet' disabled />
    </RadioGroup>
  )
}
