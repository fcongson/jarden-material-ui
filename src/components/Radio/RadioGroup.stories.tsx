import * as React from 'react'
import { Radio } from './Radio'
import { RadioGroup } from './RadioGroup'
import { useState } from 'react'

export default {
  title: 'Radio Group',
  component: RadioGroup
}

export const Default = () => {
  const [value, setValue] = useState()
  return (
    <RadioGroup name='group' value={value} onChange={value => setValue(value)}>
      <Radio label='Lorem' value='Lorem' />
      <Radio label='Ipsum' value='Ipsum' />
      <Radio label='Dolor' value='Dolor' />
      <Radio label='Sit' value='Sit' />
      <Radio label='Amet' value='Amet' />
    </RadioGroup>
  )
}
