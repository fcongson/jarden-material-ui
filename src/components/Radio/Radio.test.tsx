import { shallow } from 'enzyme'
import * as React from 'react'
import { Radio } from './Radio'

describe('Radio', () => {
  it('Renders', () => {
    const radio = shallow(<Radio />)
    expect(radio).toMatchSnapshot()
  })
})
