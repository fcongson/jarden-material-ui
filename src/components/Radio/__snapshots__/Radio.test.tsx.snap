// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Renders 1`] = `ShallowWrapper {}`;
